openquake/hazardlib/cross_correlation.py

```diff
@@ -89,3 +92,121 @@ def get_correlation_mtx(corr_model: CrossCorrelation,
                      for imt in target_imts])
     corr = np.matlib.repmat(np.squeeze(corr), num_sites, 1).T
     return corr
+
+
+# ######################## CrossCorrelationBetween ########################## #
+
+class CrossCorrelationBetween(ABC):
+    def __init__(self, trunclevel=None):
+        self.trunclevel = trunclevel
+        if self.trunclevel is None:
+            self.distribution = stats.norm()
+        elif self.trunclevel == 0:
+            self.distribution = None
+        else:
+            self.distribution = stats.truncnorm(-trunclevel, trunclevel)
+
+    @abstractmethod
+    def get_correlation(self, from_imt: IMT, to_imt: IMT) -> float:
+        """
+        :param from_imt:
+            An intensity measure type
+        :param to_imt:
+            An intensity measure type
+        :return: a scalar
+        """
+    @abstractmethod
+    def get_inter_eps(self, imts, num_events):
+        pass
+
+
+class GodaAtkinson2009(CrossCorrelationBetween):
+    """
+    Implements the correlation model of Goda and Atkinson published in 2009
+    https://pubs.geoscienceworld.org/ssa/bssa/article-abstract/99/5/3003/342221/Probabilistic-Characterization-of-Spatially
+    """
+    cache = {}  # periods -> correlation matrix
+
+    def get_correlation(self, from_imt: IMT, to_imt: IMT) -> float:
+        """
+        :returns: a scalar in the range 0..1
+        """
+        if from_imt == to_imt:
+            return 1
+
+        T1 = from_imt.period or 0.05  # for PGA
+        T2 = to_imt.period or 0.05  # for PGA
+
+        Tmin = min(T1, T2)
+        Tmax = max(T1, T2)
+        ITmin = 1. if Tmin < 0.25 else 0.
+
+        theta1 = 1.374
+        theta2 = 5.586
+        theta3 = 0.728
+
+        angle = np.pi/2. - (theta1 + theta2 * ITmin * (Tmin / Tmax) ** theta3 *
+                            np.log10(Tmin / 0.25)) * np.log10(Tmax / Tmin)
+        delta = 1 + np.cos(-1.5 * np.log10(Tmin / Tmax))
```

Comment:
```py
delta = 1 + np.cos(-1.5 * np.log10(Tmax / Tmin))
```
Eq. 8 from Goda and Atkinson (2009): 
![m_3003eq30](https://user-images.githubusercontent.com/1707414/141122941-aa437055-c5e4-49fc-8229-868113a2bf94.jpeg)

Reply:
Argh!